File: packages/javascript/lib/node/os/getDiskSize.ts
import { cwd } from 'node:process';
import { fileSizeFormat } from '../../format/fileSizeFormat.js';
import { diskSpace } from './_diskSpace.js';

export async function getDiskSize(path: string = cwd()): Promise<string> {
	return fileSizeFormat((await diskSpace(path)).total, 0, true);
}

File: packages/python/qsu/os/getDiskSize.py
import os
import shutil
from typing import Optional

from ..format.fileSizeFormat import fileSizeFormat


def getDiskSize(path: Optional[str] = None) -> str:
	# The filesystem error is raised as it is, so `errno` and `filename` survive for
	# the caller to read.
	return fileSizeFormat(shutil.disk_usage(path or os.getcwd()).total, 0, True)
